test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -335,8 +335,8 @@ def fixture_dist_gpu_backend(request):
 
 
 @pytest.fixture(autouse=True)
-def skip_by_device_type(request, instance_type):
-    is_gpu = instance_type.lstrip("ml.")[0] in ["g", "p"]
+def skip_by_device_type(request, use_gpu, instance_type):
+    is_gpu = use_gpu or instance_type.lstrip("ml.")[0] in ["g", "p"]
```

Comment:
When running SM Local tests, `instance_type` isn't `ml.a.bxlarge` but rather `local` or `local_gpu`, which leads to `is_gpu` being `False` even when the SM Local tested DLC is a GPU image on a GPU instance.